embassy-stm32/src/dma/bdma.rs

```diff
@@ -164,88 +87,87 @@ pub(crate) unsafe fn init() {
 
 pac::dma_channels! {
     ($channel_peri:ident, $dma_peri:ident, bdma, $channel_num:expr, $dmamux:tt) => {
-        impl crate::dma::sealed::Channel for crate::peripherals::$channel_peri {}
-        impl Channel for crate::peripherals::$channel_peri
-        {
-            type ReadFuture<'a> = impl Future<Output = ()> + 'a;
-            type WriteFuture<'a> = impl Future<Output = ()> + 'a;
-
-            fn read<'a>(
-                &'a mut self,
-                request: Request,
-                src: *mut u8,
-                buf: &'a mut [u8],
-            ) -> Self::ReadFuture<'a> {
-                unsafe {
-                    do_transfer(
-                        crate::pac::$dma_peri,
-                        $channel_num,
-                        (dma_num!($dma_peri) * 8) + $channel_num,
-                        request,
-                        vals::Dir::FROMPERIPHERAL,
-                        src,
-                        buf.as_mut_ptr(),
-                        buf.len(),
-                        true,
-                        #[cfg(dmamux)]
-                        <Self as super::dmamux::sealed::MuxChannel>::DMAMUX_REGS,
-                        #[cfg(dmamux)]
-                        <Self as super::dmamux::sealed::MuxChannel>::DMAMUX_CH_NUM,
-                    )
-                }
+        impl crate::dma::sealed::Channel for crate::peripherals::$channel_peri {
+
+            unsafe fn start_write<W: Word>(&mut self, request: Request, buf: &[W], reg_addr: *mut u32) {
+                low_level_api::reset_status(crate::pac::$dma_peri, $channel_num);
```

Comment:
`reset_status` could be done by `start_trasnfer`, so that there's less code inside the macro